### ISSUE TITLE
Änderung der IDs nicht möglich

### ISSUE THREAD

**Issue by larapohle** [iqb-berlin/verona-modules-aspect#712]
Lieber Achim, lieber Richard, 

wie im Feature-Forum erläutert, kann ich die IDs der likert-rows bei CmC-Items nicht ändern. Hier ein Beispiel: https://www.iqb-studio.de/#/a/558/65977/preview
Ich erhalte die Meldung "ID bereits vergeben". 

Wir können daher nicht testen, ob die Kodierung bzw. das Kodierschema funktioniert (wobei das scheinbar ohnehin noch nicht geht, weil der Schemer das noch nicht kann?) --> Das ist ja das eigentlich relevante Feature; wir wollen im besten Fall die Variablen umbenennen, ohne dass sie die Kodierung "verloren" geht.

**Comment by rhenck** [comment]
Weißt du noch, wie diese Aufgabe entstanden ist? Wurde die neu angelegt oder eine existierende auf die neue Version aktualisiert?

Kodierung und Kodieschema ist kein Aspect-Modul. Darüber hinaus unterstützt das Studio erst in der nächsten Version die Änderungen der IDs.

**Comment by larapohle** [comment]
Ich habe die Aufgabe neu angelegt.

**Comment by rhenck** [comment]
Kannst du mal bitte Berechtigungen überprüfen? Wir können die Aufgabe nicht sehen.

**Comment by larapohle** [comment]
Jetzt sollte es gehen.

**Comment by rhenck** [comment]
Okay Problem gefunden. Die IDs von Kindern in Verbundelementen wurden beim Einlesen nicht registriert. Die irreführende Fehlermeldung wurde ebenfalls korrigiert.

**Comment by franikowsp** [comment]
Ein (scheinbar anderer) Fall, bei dem uns das eben noch untergekommen ist, ist der folgende (kann es nur aus zweiter Hand wiedergeben): 

https://www.iqb-studio.de/#/a/349/47871/editor

1. Es wurde eine Umbenennung mit bestehenden Items durchgeführt (also neuer Alias vergeben).
2. Dann wurde anschließend ein Abschnitt wohl zweimal kopiert (der Abschnitt, der ein Item mit dem Alias 04 beinhaltete).
3. Anschließend gab es drei Abschnitte mit dem Alias 04. Es wird nur einmal das Item 04 in der Kodierungsliste aufgeführt (welches, ist natürlich unklar).
4. Eine Umbenennung des Alias der betroffenen Variablen ist nicht mehr möglich, was kritisch ist.

**Comment by rhenck** [comment]
> Ein (scheinbar anderer) Fall, bei dem uns das eben noch untergekommen ist, ist der folgende (kann es nur aus zweiter Hand wiedergeben):
> 
> https://www.iqb-studio.de/#/a/349/47871/editor
> 
>     1. Es wurde eine Umbenennung mit bestehenden Items durchgeführt (also neuer Alias vergeben).
> 
>     2. Dann wurde anschließend ein Abschnitt wohl zweimal kopiert (der Abschnitt, der ein Item mit dem Alias 04 beinhaltete).
> 
>     3. Anschließend gab es drei Abschnitte mit dem Alias 04. Es wird nur einmal das Item 04 in der Kodierungsliste aufgeführt (welches, ist natürlich unklar).
> 
>     4. Eine Umbenennung des Alias der betroffenen Variablen ist nicht mehr möglich, was kritisch ist.

Wenn du die Zeit hast, probier es gerne nochmal in der neuen Beta (4). Ansonsten schaue ich demnächst selber mal drauf.

**Comment by rhenck** [comment]
Hm sollte eigentlich schon zu retten sein, aber kann gerade nicht sagen, was da kaputt ist. Da muss ich mal genauer draufschauen.

**Comment by franikowsp** [comment]
Okay, dann versuchen wir das erstmal. Habe mir gerade die Unit auch einmal exportiert und scheinbar wurden auch die IDs kopiert, nicht nur die Aliase.

**Comment by rhenck** [comment]
Okay also das Problem habe ich gefunden und bereinigt. Da fehlten noch Dinge beim Initialisieren von Lückentextkindern.

Ich habe noch nicht verstanden, wie die doppelten Aliase entstanden sind. War das eine bestehende Aufgabe, die hochgezogen wurde?

**Comment by franikowsp** [comment]
Ja, die 04 wurde unter beta2 angelegt - vor dem Update auf beta3. Danach ließ sich nichts mehr umbenennen.

**Comment by rhenck** [comment]
Gibt jetzt Beta 5, die Probleme mit Lückentexten beheben sollte.

Kaputte Aufgaben kann und wird die nicht reparieren. Wenn es aber nur wenige Aufgaben sind, können wir manuell die IDs ändern.

**Comment by larapohle** [comment]
Funktioniert nun ebenfalls. Danke!
